### PR TITLE
Fix incorrect url of searching repo by language in user profile page when user have `.profile` repository

### PR DESCRIPTION
**Pull request by yp05327**
Caused by `overview` tabs (custom profile page). If user have `.profile` repo, then `overview` tab will become the default tab instead of `repository` tab.
And `$.Link` does not have query info, so we will search in overview tab by default but not in repository tab which is not expected, so we just need to add `?tab=xxx` in the query.

Before:
image
If user doesn't have `.profile` repo, it works well. But if user has, it will redirect to the wrong page:
image

After:
image

### CODE REVIEW

**Comment by yp05327** [comment]
emmm, in org home page we will get this url:
image

**Comment by CaiCandong** [comment]
Before:
before
After:
After

File: templates/explore/repo_list.tmpl [hunk 1]
```diff
@@ -40,7 +40,7 @@
 					</div>
 					<div class="flex-item-trailing">
 						{{if .PrimaryLanguage}}
-							<a class="muted" href="{{$.Link}}?q={{$.Keyword}}&sort={{$.SortType}}&language={{.PrimaryLanguage.Language}}">
+							<a class="muted" href="{{$.Link}}?{{if $.PageIsUserProfile}}tab={{$.TabName}}&{{end}}q={{$.Keyword}}&sort={{$.SortType}}&language={{.PrimaryLanguage.Language}}">
```
Comment (CaiCandong) [inline templates/explore/repo_list.tmpl:31]:
```suggestion
							<a class="muted" href="{{$.Link}}?tab={{$.TabName}}&q={{$.Keyword}}&sort={{$.SortType}}&language={{.PrimaryLanguage.Language}}">
```

Comment (yp05327) [reply templates/explore/repo_list.tmpl:31]: If it is not user profile page, we will always have a strange `tab` param in the url.

Comment (CaiCandong) [reply templates/explore/repo_list.tmpl:31]: This bug is similar to #26374,Without a user profile page, it's not surprising that `tab=repositories` when you visit repositories.

Comment (yp05327) [reply templates/explore/repo_list.tmpl:31]: I don't think this is right.
e.g. We have no `repositories` tab in the following page.
yourdomain/notifications/watching
image

Comment (yp05327) [reply templates/explore/repo_list.tmpl:31]: ops, this page can not filter repo by language. We need to fix this in another PR.

**Comment by wxiaoguang** [comment]
->  Make profile page work with repositories list #26475

File: templates/explore/repo_list.tmpl (hunk 1 above)
Comment (yp05327) [reply templates/explore/repo_list.tmpl:31]: how can we access `/notifications/subscriptions` from web?
I found this page in the source code, but have no idea about how to access it from web.

Edited:
I got it.
image

**Comment by puni9869** [comment]
I this we should backport this.

**Comment by yp05327** [comment]
This has been fixed by #28320.

**Comment by delvh** [comment]
Oops

**Pull request closed without merging**